tentacle/src/transports/tcp_base_listen.rs

```diff
@@ -0,0 +1,488 @@
+) {
+    loop {
+        match upgrade_mode {
+            UpgradeModeEnum::OnlyTcp => {
+                if sender
+                    .send((
+                        socketaddr_to_multiaddr(remote_address),
+                        MultiStream::Tcp(stream),
+                    ))
+                    .await
+                    .is_err()
+                {
+                    debug!("receiver closed unexpectedly")
+                }
+                return;
+            }
+            #[cfg(feature = "ws")]
+            UpgradeModeEnum::OnlyWs => {
+                match crate::runtime::timeout(timeout, accept_async(stream)).await {
+                    Err(_) => debug!("accept websocket stream timeout"),
+                    Ok(res) => match res {
+                        Ok(stream) => {
+                            let mut addr = socketaddr_to_multiaddr(remote_address);
+                            addr.push(Protocol::Ws);
+                            if sender
+                                .send((addr, MultiStream::Ws(Box::new(WsStream::new(stream)))))
+                                .await
+                                .is_err()
+                            {
+                                debug!("receiver closed unexpectedly")
+                            }
+                        }
+                        Err(err) => {
+                            debug!("accept websocket stream err: {:?}", err);
+                        }
+                    },
+                }
+                return;
+            }
+            #[cfg(feature = "tls")]
+            UpgradeModeEnum::OnlyTLS => {
+                match crate::runtime::timeout(timeout, acceptor.accept(stream)).await {
+                    Err(_) => debug!("accept tls server stream timeout"),
+                    Ok(res) => match res {
+                        Ok(stream) => {
+                            let mut addr = socketaddr_to_multiaddr(remote_address);
+                            addr.push(Protocol::Tls(Cow::Borrowed("")));
+                            if sender
+                                .send((addr, MultiStream::Tls(Box::new(stream))))
+                                .await
+                                .is_err()
+                            {
+                                debug!("receiver closed unexpectedly")
+                            }
+                        }
+                        Err(err) => {
+                            debug!("accept tls server stream err: {:?}", err);
+                        }
+                    },
+                }
+                return;
+            }
+            #[cfg(feature = "tls")]
+            UpgradeModeEnum::TcpAndTLS => {
+                let mut peek_buf = [0u8; 16];
+                if let Err(e) = stream.peek(&mut peek_buf).await {
+                    debug!("stream encountered err: {}, close unexpectedly", e);
+                    return;
+                }
```

Comment:
https://docs.rs/tokio/latest/tokio/net/struct.TcpStream.html#method.peek

Do we need to be concerned about the number of bytes peeked when using `stream.peek`? I'm worried that `stream.peek(&mut peek_buf)` might not fill the entire `peek_buf`.

Reply:
I will add this check